git/working.go

```diff
@@ -89,6 +90,12 @@ func (r *Repo) Clone(ctx context.Context, conf Config) (*Checkout, error) {
 	}
 	r.mu.RUnlock()
 
+	if _, err := os.Stat(repoDir + "/.gitsecret"); err == nil && conf.GitSecret {
```

Comment:
I would make the `.gitsecret` folder configurable, as by looking at the git-secret documentation, the user can control this by setting an `$SECRETS_DIR` environment variable.

This would require you to add `SECRETS_DIR` to (and maybe also `SECRETS_EXTENSION`) https://github.com/weaveworks/flux/blob/86eed4de40d539704862505a6d912726b323176c/git/operations.go#L28

In addition, I do not think I would check if the directory is there as a requirement to attempt an unseal, as the user has explicitly instructed us to run it by setting the flag. Having an additional 'hidden' check may cause confusion, instead, I would just bail out on the error `secretUnseal` returns.